spec/src/main/asciidoc/configuration-class.adoc

```diff
@@ -0,0 +1,237 @@
+
+... A xref:terminology.adoc#raw-configuration-value[raw configuration value] represented as a particular Java type, or
+
+... A configuration represented as its configuration class
++
+NOTE: If a `public` instance method declared by a configuration class returns an instance of that very configuration
+class, non-portable behavior results. (Cycles are not permitted.)
+
++
+[NOTE]
+====
+In this revision of the specification, (a) representation of xref:terminology.adoc#raw-configuration-value[raw
+configuration values] and (b) conversion of values from their representation in a particular
+xref:terminology.adoc#configuration-format[configuration format] to a partiuclar Java type have not yet been specified.
+====
+
+==== Examples
+
+This section is non-normative.
+
+NOTE: All terminology in this section is defined in the xref:terminology.adoc#terminology[Terminology] section.
+
+[source,java]
+----
+public interface MyComponent { // <1>
+
+    public String name(); // <2>
+
+    public int size(); // <3>
+
+    public Subassembly subassembly(); // <4>
+
+    public interface Subassembly { // <5>
+
+        public java.util.List<Integer> partNumbers(); // <6> <7>
+
+    }
+
+}
+
+public interface Superstructure { // <8>
+
+    public MyComponent mainComponent(); // <9>
+
+    public MyComponent subComponent(); // <10>
+
+    public MyComponent.Subassembly subassembly(); // <11>
+
+}
+----
+<1> This configuration class, considered apart from its members, represents a configuration to which the configuration
+paths it represents are relative.
+
+<2> This method, `name()`, represents a single-element configuration path, comprising a single configuration key whose
+xref:terminology.adoc#term-canonical-representation[canonical representation] is `name`, relative to the configuration
+represented by its declaring class, that (hopefully) resolves to a `String`-typed representation of a configuration
+value.
+
+<3> This method, `size()`, represents a single-element configuration path, comprising a single configuration key whose
+canonical representation is `size`, relative to the configuration represented by its declaring class, that (hopefully)
+resolves to an `int`-typed representation of a configuration value.
+
+<4> This method, `subassembly()`, represents a single-element configuration path, comprising a single configuration key
+whose canonical representation is `subassembly`, relative to the configuration represented by its declaring class, that
+(hopefully) resolves to a `Subassembly`-typed representation of a configuration value that here happens to be a (nested)
+configuration.
+
+<5> This configuration class, considered apart from its members, and apart from its declaring class, represents a
+configuration to which the configuration paths it represents are relative.
+
+<6> This method, `partNumbers()`, considered on its own, represents a single-element configuration path, comprising a
```

Comment:
I'm not sure if the representation of the name should be in camelCase. For instance, JPA Jakarta properties use kebab-case.